Vorbis decoding no longer works following bug 241339

jyavenard opened the pull request:
#### 092367a88f872d642bff95d39b0fef220437b92d
<pre>
Vorbis decoding no longer works following bug 241339
<a href="https://bugs.webkit.org/show_bug.cgi?id=242040">https://bugs.webkit.org/show_bug.cgi?id=242040</a>
&lt;rdar://96023176&gt;

Reviewed by Per Arne Vollan.

To be able to register the vorbis decoder we need an entitlement now that
we&apos;ve switched to using AudioComponentFetchServerRegistrations.

Covered by existing tests, however those are disable as the code is non-functional
with non-signed binaries.

* Source/WebKit/GPUProcess/mac/GPUProcessMac.mm:
(WebKit::GPUProcess::initializeSandbox):
* Source/WebKit/Scripts/process-entitlements.sh:

Canonical link: <a href="https://commits.webkit.org/252211@main">https://commits.webkit.org/252211@main</a>
</pre>

pvollan reviewed (approved):
R=me.

pvollan commented on Source/WebKit/GPUProcess/mac/GPUProcessMac.mm:
```diff
@@ -72,6 +73,11 @@
     // Need to overide the default, because service has a different bundle ID.
     NSBundle *webKit2Bundle = [NSBundle bundleForClass:NSClassFromString(@"WKWebView")];
 
+#if defined(USE_VORBIS_AUDIOCOMPONENT_WORKAROUND)
```
This could also be written as: #if USE(VORBIS_AUDIOCOMPONENT_WORKAROUND)

jyavenard committed:
Vorbis decoding no longer works following bug 241339
https://bugs.webkit.org/show_bug.cgi?id=242040
<rdar://96023176>

Reviewed by Per Arne Vollan.

To be able to register the vorbis decoder we need an entitlement now that
we've switched to using AudioComponentFetchServerRegistrations.

Covered by existing tests, however those are disable as the code is non-functional
with non-signed binaries.

* Source/WebKit/GPUProcess/mac/GPUProcessMac.mm:
(WebKit::GPUProcess::initializeSandbox):
* Source/WebKit/Scripts/process-entitlements.sh:

Canonical link: https://commits.webkit.org/252211@main
